### PR TITLE
#14688 Scan the repo with clang-tidy as part of post-commit

### PR DESCRIPTION
**Pull request by afuller-TT**
### Ticket
#14688 

### Problem description
To leverage clang-tidy we need assurance that the entire repo conforms to the specified clang-tidy config.

### What's changed
Added a scan in post-commit and fixed up the repo as needed for the scan to come up clean.
Now we have a baseline to work from.

### Checklist
- [x] Post commit CI passes https://github.com/tenstorrent/tt-metal/actions/runs/11853360350
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by blozano-tt — APPROVED** [review]
LGTM! Nice work!

**Comment by dmakoviichuk-tt** [comment]
@afuller-TT I'd like to have our own clang-tidy file for now and unify it with main file later.

File: CMakePresets.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (ayerofieiev-tt) [inline CMakePresets.json:1]: 👍

**Comment by afuller-TT** [comment]
> @afuller-TT I'd like to have our own clang-tidy file for now and unify it with main file later.

That's fine @dmakoviichuk-tt if you want.  However the reason I did this was because the config that was in place was not clean to the repo.  Issues were detected and were failing the whole-repo scan that I'm implementing here.

So to get your own config file you'll need to find out which checks need to be disabled in order to get a clean bill of health.

You'll also miss out on whole-repo sweeps to address issues.  I have a branch waiting in the wings that cleans up all `performance-for-range-copy` for example.  If a subdir like `tt-train` has its own config file, then a root-level sweep will not clean up anything in that subdir and you'll miss out on free work.

Up to you.  But if you want your own config immediately, then I ask that you put in a config file that `tt-train` is clean to so that we can activate this scan sooner rather than later.

**Comment by dmakoviichuk-tt** [comment]
@afuller-TT okay lets put the same copy your clang-tidy file in our repo. I want to be able to add more rules and own my own file. We have smaller codebase I believe I can introduce stricter rules faster.

**Comment by afuller-TT** [comment]
> @afuller-TT okay lets put the same copy your clang-tidy file in our repo. I want to be able to add more rules and own my own file. We have smaller codebase I believe I can introduce stricter rules faster.

As you wish.

**Review by ttmchiou — APPROVED** [review]

**Review by dmakoviichuk-tt — APPROVED** [review]

**Pull request merged**